### PR TITLE
Update target frameworks

### PR DESCRIPTION
**Pull request by NeilSchiefer**
## Description

Updated target frameworks to support only net6.0, netcoreapp3.1, and net48.

Added editorconfig and Directory.Build.props for consistency

## Type of change: 
This is a breaking change and updated version to 2.0.0.

---

<sup>_[Reviewer guidelines](../blob/main/CONTRIBUTING.md#reviewing-changes)_</sup>

### CODE REVIEW

File: RockLib.Compression.Tests/RockLib.Compression.Tests.csproj [hunk 1]
```diff
@@ -1,14 +1,14 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFrameworks>net5.0;netcoreapp3.1;net462;</TargetFrameworks>
+    <TargetFrameworks>net6.0;netcoreapp3.1;net48;</TargetFrameworks>
```
Comment (alpacamybags118) [inline RockLib.Compression.Tests/RockLib.Compression.Tests.csproj]: does this have any impact on what version of dotnet the consumer needs to be using? if so, that should probably be documented in the README

Comment (alpacamybags118) [reply RockLib.Compression.Tests/RockLib.Compression.Tests.csproj]: (well this is the test file, but i think the main proj is updated too :) )

File: RockLib.Compression/CompressionExtensions.cs [hunk 2]
```diff
@@ -14,9 +15,11 @@ public static class CompressionExtensions
         /// <param name="compressor">Compressor to extend</param>
         /// <param name="text">The text to compress</param>
         /// <param name="encoding">The optional encoding to use.</param>
-        /// <returns></returns>
+        /// <returns>The compressed byte array.</returns>
```
Comment (alpacamybags118) [inline RockLib.Compression/CompressionExtensions.cs:18]: 👍

File: RockLib.Compression.Tests/RockLib.Compression.Tests.csproj (hunk 1 above)
Comment (NeilSchiefer) [reply RockLib.Compression.Tests/RockLib.Compression.Tests.csproj]: 👍

File: RockLib.Compression/CompressionExtensions.cs [hunk 3]
```diff
@@ -14,9 +15,11 @@ public static class CompressionExtensions
...
+        /// <exception cref="ArgumentNullException"></exception>
         public static byte[] Compress(this ICompressor compressor, string text, Encoding encoding = null)
         {
+            if (compressor == null) throw new ArgumentNullException(nameof(compressor));
```
Comment (JasonBock) [inline RockLib.Compression/CompressionExtensions.cs]: We should start using `is` for `null` comparisons, like this:
```
if (compressor is null)
```
Using `==` is fine, but it calls that operator, and that can be changed in a type definitions. It's really rare to see anyone ever change it to do something nefarious with `null`, but using `is` will always do a direct `null` comparison.

File: RockLib.Compression/IDecompressor.cs [hunk 4]
```diff
@@ -12,6 +13,7 @@ public interface IDecompressor
         /// </summary>
         /// <param name="inputStream">The stream to decompress.</param>
         /// <returns>The decompressed byte array.</returns>
+        /// <exception cref="ArgumentNullException"></exception>
```
Comment (JasonBock) [inline RockLib.Compression/IDecompressor.cs]: The method on the interface itself doesn't throw `ArgumentNullException`. So...I'm not sure we should be putting this into the comments. It's possible that someone could implement this and not throw that exception.

**Review by JasonBock — CHANGES_REQUESTED** [review]

File: RockLib.Compression/IDecompressor.cs (hunk 4 above)
Comment (NeilSchiefer) [reply RockLib.Compression/IDecompressor.cs]: Ah yeah good point.

File: RockLib.Compression/CompressionExtensions.cs (hunk 3 above)
Comment (NeilSchiefer) [reply RockLib.Compression/CompressionExtensions.cs]: 👍

File: RockLib.Compression.Tests/RockLib.Compression.Tests.csproj (hunk 1 above)
Comment (NeilSchiefer) [reply RockLib.Compression.Tests/RockLib.Compression.Tests.csproj]: updated.

File: RockLib.Compression/CompressionExtensions.cs (hunk 3 above)
Comment (NeilSchiefer) [reply RockLib.Compression/CompressionExtensions.cs]: changed.

File: RockLib.Compression/IDecompressor.cs (hunk 4 above)
Comment (NeilSchiefer) [reply RockLib.Compression/IDecompressor.cs]: removed.

**Review by eslutz — APPROVED** [review]
All the changes look good to me!

**Review by JasonBock — APPROVED** [review]

**Pull request merged**